Non-critical error trying to get screenshot even runs an api only

[INFO] tag.pagefactory.mobile.drivers.MobileDriverService - Capabilities are Capabilities {app: , appActivity: , appPackage: , appiumVersion: , autoGrantPermissions: true, browserName: , deviceName: , deviceOrientation: , platformName: , platformVersion: , resetKeyboard: true, unicodeKeyboard: true}
[ERROR] ru.sbtqa.tag.pagefactory.allure.ErrorHandler       - Can't attach screenshot to allure reports
ru.sbtqa.tag.pagefactory.exceptions.FactoryRuntimeException: Could not parse appium url. Check 'appium.url' property
	at ru.sbtqa.tag.pagefactory.mobile.drivers.MobileDriverService.mountDriver(MobileDriverService.java:53)
	at ru.sbtqa.tag.pagefactory.mobile.drivers.MobileDriverService.getDriver(MobileDriverService.java:77)
	at ru.sbtqa.tag.pagefactory.mobile.drivers.MobileDriverService.getDriver(MobileDriverService.java:1)
	at ru.sbtqa.tag.pagefactory.utils.ScreenshotUtils$1.take(ScreenshotUtils.java:21)
	at ru.sbtqa.tag.pagefactory.allure.ErrorHandler.attachScreenshot(ErrorHandler.java:20)
	at ru.sbtqa.tag.pagefactory.aspects.CriticalStepCheckAspect.attachError(CriticalStepCheckAspect.java:84)
	at ru.sbtqa.tag.pagefactory.aspects.CriticalStepCheckAspect.ajc$inlineAccessMethod$ru_sbtqa_tag_pagefactory_aspects_CriticalStepCheckAspect$ru_sbtqa_tag_pagefactory_aspects_CriticalStepCheckAspect$attachError(CriticalStepCheckAspect.java:1)
	at ru.sbtqa.tag.pagefactory.aspects.CriticalStepCheckAspect.runStep(CriticalStepCheckAspect.java:70)
	at cucumber.runtime.StepDefinitionMatch.runStep(StepDefinitionMatch.java:40)
	at cucumber.api.TestStep.executeStep_aroundBody2(TestStep.java:102)
	at cucumber.api.TestStep$AjcClosure3.run(TestStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.executeStep(StashAspect.java:67)
	at cucumber.api.TestStep.executeStep(TestStep.java:101)
	at cucumber.api.TestStep.run_aroundBody0(TestStep.java:83)
	at cucumber.api.TestStep$AjcClosure1.run(TestStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.run(StashAspect.java:52)
	at cucumber.api.TestStep.run(TestStep.java:78)
	at cucumber.api.TestCase.run(TestCase.java:58)
	at cucumber.runner.Runner.runPickle(Runner.java:80)
	at cucumber.runtime.junit.PickleRunners$NoStepDescriptions.run(PickleRunners.java:140)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:68)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:23)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:73)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:118)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:56)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at cucumber.api.junit.Cucumber$1.evaluate(Cucumber.java:127)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: java.net.MalformedURLException: no protocol: 
	at java.net.URL.<init>(URL.java:593)
	at java.net.URL.<init>(URL.java:490)
	at java.net.URL.<init>(URL.java:439)
	at ru.sbtqa.tag.pagefactory.mobile.drivers.MobileDriverService.mountDriver(MobileDriverService.java:51)
	... 45 more
    * ? system returns "error" with parameters       # ApiStepDefs.validate(String,DataTable)
      | message | Access token is missing or invalid | 
      | http code | 401 | 
      | code | 38 | 
      ru.sbtqa.tag.api.exception.RestPluginException: Failed to execute validation rule "error" in "Login" endpoint entry
	at ru.sbtqa.tag.api.utils.ReflectionUtils.invoke(ReflectionUtils.java:46)
	at ru.sbtqa.tag.api.EndpointEntryReflection.validate(EndpointEntryReflection.java:144)
	at ru.sbtqa.tag.api.EndpointEntry.validate(EndpointEntry.java:159)
	at ru.sbtqa.tag.stepdefs.en.ApiStepDefs.validate(ApiStepDefs.java:50)
	at ✽.system returns "error" with parameters(src/test/resources/features/login/Login.feature:16)
Caused by: java.lang.AssertionError: 1 expectation failed.
JSON path message doesn't match.
Expected: Access token is missing or invalid
  Actual: Account with these credentials does not exist

	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.codehaus.groovy.reflection.CachedConstructor.invoke(CachedConstructor.java:83)
	at org.codehaus.groovy.reflection.CachedConstructor.doConstructorInvoke(CachedConstructor.java:77)
	at org.codehaus.groovy.runtime.callsite.ConstructorSite$ConstructorSiteNoUnwrap.callConstructor(ConstructorSite.java:84)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallConstructor(CallSiteArray.java:60)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:235)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callConstructor(AbstractCallSite.java:247)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure.validate(ResponseSpecificationImpl.groovy:483)
	at io.restassured.internal.ResponseSpecificationImpl$HamcrestAssertionClosure$validate$1.call(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.validateResponseIfRequired(ResponseSpecificationImpl.groovy:655)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210)
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:166)
	at io.restassured.internal.ResponseSpecificationImpl.content(ResponseSpecificationImpl.groovy:275)
	at io.restassured.specification.ResponseSpecification$content$1.callCurrent(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
	at io.restassured.specification.ResponseSpecification$content$1.callCurrent(Unknown Source)
	at io.restassured.internal.ResponseSpecificationImpl.body(ResponseSpecificationImpl.groovy:258)
	at io.restassured.internal.ValidatableResponseOptionsImpl.body(ValidatableResponseOptionsImpl.java:274)
	at com.savl.qa.rest.LoginEndpointEntry.validateError(LoginEndpointEntry.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at ru.sbtqa.tag.api.utils.ReflectionUtils.invoke(ReflectionUtils.java:36)
	at ru.sbtqa.tag.api.EndpointEntryReflection.validate(EndpointEntryReflection.java:144)
	at ru.sbtqa.tag.api.EndpointEntry.validate(EndpointEntry.java:159)
	at ru.sbtqa.tag.stepdefs.en.ApiStepDefs.validate(ApiStepDefs.java:50)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at cucumber.runtime.Utils$1.call(Utils.java:32)
	at cucumber.runtime.Timeout.timeout(Timeout.java:16)
	at cucumber.runtime.Utils.invoke(Utils.java:26)
	at cucumber.runtime.java.JavaStepDefinition.execute(JavaStepDefinition.java:37)
	at cucumber.runtime.StepDefinitionMatch.runStep_aroundBody0(StepDefinitionMatch.java:40)
	at cucumber.runtime.StepDefinitionMatch$AjcClosure1.run(StepDefinitionMatch.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.CriticalStepCheckAspect.runStep(CriticalStepCheckAspect.java:67)
	at cucumber.runtime.StepDefinitionMatch.runStep(StepDefinitionMatch.java:40)
	at cucumber.api.TestStep.executeStep_aroundBody2(TestStep.java:102)
	at cucumber.api.TestStep$AjcClosure3.run(TestStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.executeStep(StashAspect.java:67)
	at cucumber.api.TestStep.executeStep(TestStep.java:101)
	at cucumber.api.TestStep.run_aroundBody0(TestStep.java:83)
	at cucumber.api.TestStep$AjcClosure1.run(TestStep.java:1)
	at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
	at ru.sbtqa.tag.pagefactory.aspects.StashAspect.run(StashAspect.java:52)
	at cucumber.api.TestStep.run(TestStep.java:78)
	at cucumber.api.TestCase.run(TestCase.java:58)
	at cucumber.runner.Runner.runPickle(Runner.java:80)
	at cucumber.runtime.junit.PickleRunners$NoStepDescriptions.run(PickleRunners.java:140)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:68)
	at cucumber.runtime.junit.FeatureRunner.runChild(FeatureRunner.java:23)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at cucumber.runtime.junit.FeatureRunner.run(FeatureRunner.java:73)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:118)
	at cucumber.api.junit.Cucumber.runChild(Cucumber.java:56)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at cucumber.api.junit.Cucumber$1.evaluate(Cucumber.java:127)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

--- a/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/aspects/CriticalStepCheckAspect.java
+++ b/page-factory-core/src/main/java/ru/sbtqa/tag/pagefactory/aspects/CriticalStepCheckAspect.java
@@ -27,10 +27,10 @@ import org.aspectj.lang.annotation.Pointcut;
 import ru.sbtqa.tag.pagefactory.allure.CategoriesInjector;
 import ru.sbtqa.tag.pagefactory.allure.Category;
 import ru.sbtqa.tag.pagefactory.allure.ErrorHandler;
+import ru.sbtqa.tag.pagefactory.environment.Environment;
 import ru.sbtqa.tag.pagefactory.exceptions.AllureNonCriticalError;
 import ru.sbtqa.tag.pagefactory.exceptions.ReadFieldError;
 import ru.sbtqa.tag.pagefactory.optional.PickleStepCustom;
-import ru.sbtqa.tag.pagefactory.utils.Wait;
 import ru.sbtqa.tag.qautils.errors.AutotestError;
 
 @Aspect
@@ -82,7 +82,9 @@ public class CriticalStepCheckAspect {
     private void attachError(PickleStepCustom step, Throwable e) {
         step.setError(e);
         ErrorHandler.attachError(e);
-        ErrorHandler.attachScreenshot();
+        if (!Environment.isDriverEmpty()) {
+            ErrorHandler.attachScreenshot();
+        }
         CategoriesInjector.inject(nonCriticalCategory);
     }
 

